src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -250,8 +250,9 @@ def build_vm_resource(  # pylint: disable=too-many-locals
         disk_info=None, boot_diagnostics_storage_uri=None, ultra_ssd_enabled=None):
 
     os_caching = disk_info['os'].get('caching')
-    # TODO: split out the storage_sku for os disk(can't be ultra-ssd) and individual data disks
-    os_storage_sku = None if storage_sku == 'UltraSSD_LRS' else storage_sku
+    os_storage_sku = disk_info['os'].get('storageAccountType')
+    if os_storage_sku and os_storage_sku.lower() == 'UltraSSD_LRS'.lower():
+        os_storage_sku = None
```

Comment:
I might get lost here, but to me we should either error or warn here, and the code should be moved to validators

Reply:
Nit: `'UltraSSD_LRS'.lower()` can be `ultrassd_lrs`